crates/sui-graphql-rpc/src/types/coin.rs

```diff
@@ -408,5 +404,7 @@ fn apply_filter(mut query: RawQuery, coin_type: &TypeTag, owner: Option<SuiAddre
         coin_type.to_canonical_display(/* with_prefix */ true)
     );
 
-    query
+    // Since we have partial indexes on `coin_type IS NOT NULL` and `owner_type = 1`, every coin
+    // query will have this filter condition.
+    filter!(query, format!("owner_type = {}", OwnerType::Address as i16))
```

Comment:
I don't understand the correctness argument for this filter. Coins can be shared or frozen -- it's a strange thing to do to a Coin, but it's possible, and if you did it, you would "hide" it from the perspective of RPC, which we do not want to do.

Reply:
In that case we should drop the partial index on owner_type = 1 then. The "correctness" is because of the partial index, so when querying for coins without using the same filtering condition in our query, we wouldn't end up using it

Reply:
I still don't quite understand this after reading the explanation. If there is a shared coin of some type, this filter would add `owner_type = address` to the query as well and the shared coin would not show up. Why is this right?

Reply:
We would drop the partial index condition, and for queries, drop the filtering condition on owner_type as well